### PR TITLE
3.8.2 release

### LINKED ISSUE #2066: Release rspec-rails 3.8.2
Given that there aren't any new features merged in to master, let's make sure we've forward-ported all the bugfixes and enhancements from `master` to  `3-8-maintenance` and release `3.8.2` so that we can then begin the process of developing RSpec Rails 4.

@benoittgt do you want to tackle this issue?

### PR DESCRIPTION
**Pull request by benoittgt**
This resolves #2066 

Because this is my first time, I prefere to open a pull request to validate step by step. 😊

Checklist:
- [x] Check out `3-8-maintenance`
- [x] Cherry pick bug fixes that aren't on `3-8-maintenance`, editing changefile as necessary
- [x] Finalise changelog, bump version in [lib/rspec/rails/version.rb](https://github.com/rspec/rspec-rails/blob/master/lib/rspec/rails/version.rb) and commit "Update version to x.x.x"
- [x] Git tag to new version
- [ ] Run `rake release` with an account that has credentials logged in
- [ ] Ensure fully pushed to `3-8-maintenance`
- [ ] Check out master and ensure change log is updated. (This could be a PR from maintenance).
- [ ] 🎉

### CODE REVIEW

**Comment by samphippen** [comment]
@benoittgt this looks great to me!

Is there any chance you could add this checklist to the wiki on rspec-dev, so we have it for next time? (you can use x-y-maintenance to represent major and minor, instead of 3-8-maintenance)

**Comment by samphippen** [comment]
@benoittgt this is going to be a party... you'll need the signing key for the RSpec gem, the password, and release bits on rubygems.

@JonRowe I don't have the key on the computer I'm currently travelling with. Could you email it to both me and @benoittgt?

@benoittgt do you have a GPG key? If so I'll sign and encrypt the password for the RSpec signing key and send it over to you with instructions.

@JonRowe do we have the procedure for setting your keys up documented somewhere?

File: Changelog.md [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 * Fix the request spec generator to handle namespaced files. (Kohei Sugi, #2057)
+* Further truncate system test filenames to handle cases when extra words are
+  prepended. (Takumi Kaji, #2058)
+
+Enhancements
+
+* Use `__dir__` instead of `__FILE__` in generated `rails_helper.rb` where
+  supported. (OKURA Masafumi, #2048)
+* Add `have_enqueued` matcher as a "super" matcher to the `ActiveJob` matchers
+  making it easier to match on `ActiveJob` delivered emails (Joel Lubrano, #2047)
```
Comment (JonRowe) [inline Changelog.md]: @benoittgt we don't typically release enhancements in a patch release, so these should be removed

Comment (benoittgt) [reply Changelog.md]: I remove them.

**Comment by benoittgt** [comment]
> Is there any chance you could add this checklist to the wiki on rspec-dev, so we have it for next time? 

@samphippen except this I don't see any github wiki of place where it could be added. 🤔 https://github.com/rspec/rspec-dev/blob/8b5a2ba438bb442861c43971f6378d5cd2a94d77/common_markdown_files/CONTRIBUTING.md.erb#L35

I just sent my PGP key to you and @JonRowe by email.

Thanks for the review!

**Comment by JonRowe** [comment]
LGTM @benoittgt I recommend you merge and then release in short order.

**Comment by JonRowe** [comment]
@benoittgt can we add #2069 to this

**Comment by benoittgt** [comment]
The problem with #2069 is that we don't add https://github.com/rspec/rspec-rails/pull/2047 in this patch release. So https://github.com/rspec/rspec-rails/pull/2069/files#diff-22e10b416305c8e9a9b607039ccfe28d is "deleted by us" when cherry picking.

I can backport this fix to work on code present in 3.8.2 and add a changelog specific for that. What do you think?

**Comment by JonRowe** [comment]
I'd backport it as a PR to your branch keep the changelog but refer to the extra PR

**Comment by benoittgt** [comment]
Ok backported. Commit order is different here https://github.com/rspec/rspec-rails/pull/2067/commits VS here https://github.com/benoittgt/rspec-rails/commits/3-8-maintenance-prep 🤔

**Comment by benoittgt** [comment]
Yes maybe. I sent a support request.

**Comment by benoittgt** [comment]
It seems ok now.

**Comment by benoittgt** [comment]
@samphippen Still interested to do https://github.com/rspec/rspec-rails/pull/2067#issuecomment-453840109 😊

**Pull request merged**